### ISSUE TITLE
Namespaces keys order

### ISSUE THREAD

**Issue by eduardocp** [renggli/dart-xml#64]
Why namespaces key and value has the key as the uri of namespace instead of the real key on xml?

Ex.: 
```dart
var namespaces = <String, String>{
    'soap' : 'http://www.w3.org/2003/05/soap-envelope/',
};
var builder = new xml.XmlBuilder();
builder.processing('xml', 'version="1.0"');
builder.element('soap:Envelop', namespaces: namespaces, nest: () { });

var result = builder.build();
```

produces: 
```xml
<?xml version="1.0"?><soap:Envelop xmlns:http://schemas.xmlsoap.org/soap/envelope/="soap"/>
```

**Comment by renggli** [comment]
You need to define the namespaces like so:

```dart
  final namespaces = {
    'http://www.w3.org/2003/05/soap-envelope/': 'soap' ,
  };
```

See the [documentation](https://github.com/renggli/dart-xml/blob/c92854b58cb3cd03df08964a89f891ebcbeedb52/lib/src/xml/builder.dart#L93).

**Comment by eduardocp** [comment]
Yes, I saw the documentation. Maybe I expressed myself badly. I wanted to understand why this is so, since it is more intuitive (in my opinion) that the key is alias not uri.

**Comment by renggli** [comment]
I see, this is not very obvious. I am trying to remember why the key-values are ordered like this? Other implementations also seem to do it this way, i.e. http://effbot.org/zone/element-namespaces.htm. Probably this has to do with the underlying implementation, but maybe there could be better ways of exposing this to users?

**Comment by rspilker** [comment]
I think the main reason is that a namespace is defined by a uri, therefore it seems a good idea to make it a key.

A namespace is not required to have an alias (prefix). Still points to a key with a null value.

But a namespace can have zero _or more aliases_ (not zero or one alias).  

Making the alias the key would solve the _or more_ problem, but not the _zero_ case.

Since it is very unlikely that a single element would use multiple aliases for a certain namespace, having the namespace as a key makes more sense.

I think instead of using a `Map<String,String?>', a separate type would have been a better option.

**Comment by rspilker** [comment]
Hmm, the signature is not `Map<String,String?)` but `Map<String,String>`. However, you could always call `void namespace(String uri, [String? prefix])` instead.

**Comment by renggli** [comment]
Yeah, the signature should be `Map<String, String?>` for consistency.

**Comment by renggli** [comment]
Fixed this with f460492c7501ac525ebfdf96456eebdf82f28a12.
